internal/provider/resource_gitlab_project_milestone_test.go

```diff
@@ -0,0 +1,178 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabProjectMilestone_basic(t *testing.T) {
+	testAccCheck(t)
+
+	var milestone gitlab.Milestone
+	var milestoneUpdate gitlab.Milestone
+	rInt1, rInt2 := acctest.RandInt(), acctest.RandInt()
+	project := testAccCreateProject(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabProjectMilestoneDestroy,
+		Steps: []resource.TestStep{
+			{
+				// create Milestone with required values only
+				Config: testAccGitlabProjectMilestoneConfigRequiredOnly(project.ID, rInt1, ""),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectMilestoneExists("this", &milestone),
+					testAccCheckGitlabProjectMilestoneAttributes("this", &milestone, &testAccGitlabProjectMilestoneExpectedAttributes{
```

Comment:
Can you use the resource helpers from `resource` ? Like `resource.TestCheckResourceAttr` etc?

Also note the testing guidelines we are working on here: https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008

e.g. non-computed field don't need to be asserted explicitly in case of a subsequent import verification.

Reply:
@timofurrer 
So, if I rightly understand [this](https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008) and [this](https://github.com/hashicorp/terraform-plugin-sdk/blob/5adf5f1c4bf0aee7bb31d51cd7f016e81a39e3e5/helper/resource/testing.go) we should use the function
`resource.TestCheckResourceAttrSet("gitlab_foo.foo", "bar"),` for fields `Computed:    true`, aren't we?

Reply:
Yes, or `resource.TestCheckResourceAttrWith` if you want to check for the value which may be dynamic, see:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/5c1b5250b38bf4e207ee4381b9c6e27e8e577f52/internal/provider/resource_gitlab_project_issue_test.go#L43

Reply:
... in case it's known before hand you could use `resource.TestCheckResourceAttr` instead.